examples/pointer_generator/pointer_generator_src/transformer_pg.py

```diff
@@ -351,18 +351,18 @@ def output_layer(
         # vocab_size]. Each attention weight will be written into a location
         # that is for other dimensions the same as in the index tensor, but for
         # the third dimension it's the value of the index tensor (the token ID).
-        attn = torch.mul(attn, 1 - p_gens)
+        attn = torch.mul(attn.float(), 1 - p_gens)
```

Comment:
curious about the `float()` call here and below (`scatter_add_`), did you folks find this important for stability?

Reply:
Yes, pointer_generator was unstable even after switching to AMP from FP16. We tried incrementally forcing ops to fp32 and found that these two in fp32 provided stable training. Another change needed was uninstalling apex so that `layer_norm` is not autocast to fp16, as `fused_layer_norm` is I think autocast to fp16 by default.